### ISSUE TITLE
Using PhantomJS

### ISSUE THREAD

**Issue by dve** [webdriverextensions/webdriverextensions#69]
I tried to use PhantomJS with the `@PhantomJS` annotation, but nothing happened - just the tests are skipped. I found out it's because  `BrowserConfiguration.createDriver()` there is just an exception thrown which leads to a `TRACE` level logging.
   * Is there a way to use PhantomJS? Then it should be documented.
   * Should I not use PhantomJS because its deprecated? The the `@PhantomJS` annotation should be deprecated too, and logging should be a least `INFO`
   * Is it's missing in `createDriver()` a bug? The we should fix it.

**Comment by andidev** [comment]
I think it's not supported to run the test locally since the createDriver is not implemented. Though it will run remotely with the  @RemoteAddress annotation present.

The idea is that every annotated driver that can be run remotely should run locally and be ignored locally if it is not supported to run locally. I guess this should be documented. 

I'm not too familiar with the phantomjs driver. Is it deprecated in selenium? If not it could be easily implemented by adding the proper create driver logic as you mention. I think also a default Driver path needs to be added to DriverPathLoader.  Please feel free to contribute with a PR if you want.

**Comment by andidev** [comment]
The log should definitely be a info level log that says that the driver is skipped locally

**Comment by dve** [comment]
phantomjs driver is not deprecated by itself, but the java binding [ghostdriver] (https://github.com/detro/ghostdriver) is not maintained. I will check out if its still working localy, and then provide a pull request, either with just the logging change or with a create driver logic.
Do you prefer a "hard" or an "optional" dependency?

**Comment by andidev** [comment]
Not sure. Would optional mean that the dependency scope is set to runtime in maven? 
What do you think?

**Comment by dve** [comment]
I mean `<optional>true</optional>` in the `dependency` tag, this would cause ghostdriver not to be included by default, users would need to declare it themselves.

**Comment by dve** [comment]
The problem is I don't got the ghostdriver running. And even if - it depends on phantomjs 1.9.

**Comment by andidev** [comment]
Aha okey did not now about the optional alternative in maven. 

Maybe I could get a look at it this weekend as well. I think @lkwg82 maybe runs selenium with phantomjs driver. @lkwg82 do you know if phantomjs actually works with selenium for java?

**Comment by andidev** [comment]
The ghostdriver does not look up to date as you say. Though 
```xml
<dependency>
    <groupId>com.codeborne</groupId>
    <artifactId>phantomjsdriver</artifactId>
    <version>1.3.0</version>
</dependency>
```
could maybe be used as a replacement.
I tried it out and it seems to work.

Here is a current working branch [add-phantomjsdriver-support](https://github.com/webdriverextensions/webdriverextensions/tree/add-phantomjsdriver-support) version 2.9.0-SNAPSHOT.
Do you use windows or linux? Could you please try it out then, since I'm on a mac. I also changed the debug levels to info for some more logs.

**Comment by dve** [comment]
Thanks, on ubuntu 16.04 the `WebDriverRunnerTest` works fine with `@PhantomJS`. I will do some more testing later and also ask someone using windows to test.

**Comment by dve** [comment]
Testing on windows failed because the driver downloader seems to have problems with windows paths:
```
INFO] Installation directory "D:\birke\webdriverextensions-master\drivers"
[INFO] Installing drivers from configuration
[INFO] internetexplorerdriver-windows-32bit version 2.53.0
[INFO]   Downloading
"http://selenium-release.storage.googleapis.com/2.53/IEDriverServer_Win32_2.53.0.zip"
to
"C:\Users\birke\AppData\Local\Temp\webdriverextensions-maven-plugin\temp\IEDriverServer_Win32_2.53.0.zip"
[INFO]   Extracting
"C:\Users\birke\AppData\Local\Temp\webdriverextensions-maven-plugin\temp\IEDriverServer_Win32_2.53.0.zip"
to temp folder
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15.088 s
[INFO] Finished at: 2016-09-27T14:06:15+02:00
[INFO] Final Memory: 19M/209M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal
com.github.webdriverextensions:webdriverextensions-maven-plugin:3.0.0:install-drivers
(default) on project webdriverextensions: Execution default of goal
com.github.webdriverextensions:webdriverextensions-maven-plugin:3.0.0:install-drivers
failed: Illegal/unsupported escape sequence near index 3
[ERROR] C:\Users\birke\AppData\Local\Temp\webdriverextensions-maven-plugin\temp\
[ERROR] ^
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e
switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please
read the following articles:
[ERROR] [Help 1]
http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

Shall I open a ticket for this?

**Comment by andidev** [comment]
This bug is fixed in version 3.0.2 of the webdriverextensions-maven-plugin-repository. So just change the version.

**Comment by dve** [comment]
It also works on windows. Thanks a lot.

**Comment by andidev** [comment]
NP! Released in version 3.9.0 available now in Maven repo.
